worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala

```diff
@@ -231,7 +231,10 @@ final private[worker] class StorageManager(conf: CelebornConf, workerSource: Abs
   def updateFileInfosInDB(): Unit = {
     fileInfos.asScala.foreach { case (shuffleKey, files) =>
       try {
-        db.put(dbShuffleKey(shuffleKey), PbSerDeUtils.toPbFileInfoMap(files))
+        // Save fileinfos that has not been persist when graceful shutdown
+        if (db.get(dbShuffleKey(shuffleKey)) != null) {
```

Comment:
Currently, LifecycleManager can send multiple CommitFiles to  a shuffle, especially for `MapPartition`, each mapper end will trigger a CommitFile, cc @RexXiong . So I think we should not add the condition here.